### PR TITLE
Add cortex-m support to cstool, refactoring and crash fix

### PR DESCRIPTION
**Pull request by plaes**

### CODE REVIEW

File: cstool/cstool.c [hunk 1]
```diff
@@ -183,145 +184,99 @@ int main(int argc, char **argv)
...
-
-	if (!strcmp(mode, "thumble")) {
+		md = CS_MODE_THUMB | CS_MODE_BIG_ENDIAN;
+	} else if (!strcmp(mode, "thumble")) {
 		arch = CS_ARCH_ARM;
-		err = cs_open(CS_ARCH_ARM, CS_MODE_ARM | CS_MODE_LITTLE_ENDIAN, &handle);
-	}
-
-	if (!strcmp(mode, "cortexm")) {
+		md = CS_MODE_ARM | CS_MODE_LITTLE_ENDIAN;
```
Comment (aquynh) [inline cstool/cstool.c]: is this a mistake, because this should be for Thumb?

File: cstool/cstool.c [hunk 2]
```diff
@@ -182,140 +184,99 @@ int main(int argc, char **argv)
...
 		arch = CS_ARCH_ARM64;
-		err = cs_open(CS_ARCH_ARM64, CS_MODE_LITTLE_ENDIAN, &handle);
-	}
-
-	if (!strcmp(mode, "arm64be")) {
+		md = CS_MODE_LITTLE_ENDIAN;
+	} else if (!strcmp(mode, "arm64be")) {
 		arch = CS_ARCH_ARM64;
+		md = CS_MODE_BIG_ENDIAN;
 		err = cs_open(CS_ARCH_ARM64, CS_MODE_BIG_ENDIAN, &handle);
```
Comment (aquynh) [inline cstool/cstool.c]: this line should be removed?

**Comment by plaes** [comment]
OK, sorry for the mess. I rearranged the commits:
1) Refactoring and crash fix
2) Add cortexm functionality

**Comment by aquynh** [comment]
merged, thanks!

would you make a similar pull req for the "next" branch, please?

**Pull request merged**